supernode/daemon/mgr/progress/gc_util.go

```diff
@@ -0,0 +1,81 @@
+package progress
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+)
+
+// DeleteTaskID deletes the super progress with specified taskID.
+func (pm *Manager) DeleteTaskID(ctx context.Context, taskID string) (err error) {
+	return pm.superProgress.remove(taskID)
+}
+
+// DeleteCID deletes the client progress with specified clientID.
+func (pm *Manager) DeleteCID(ctx context.Context, clientID string) (err error) {
+	return pm.clientProgress.remove(clientID)
+}
+
+// DeletePeerID deletes info about PeerID.
+func (pm *Manager) DeletePeerID(ctx context.Context, peerID string) error {
+	// NOTE: we should delete the peerID from the pieceProgress.
+	// However, it will cost a lot of time to find which one refers to this peerID.
+	// So we leave it to be deleted when scheduled.
+	pm.deletePeerIDFromPeerProgress(ctx, peerID)
+	pm.deletePeerIDFromBlackInfo(ctx, peerID)
+
+	return nil
+}
+
+func (pm *Manager) deletePeerIDFromPeerProgress(ctx context.Context, peerID string) bool {
+	if err := pm.peerProgress.remove(peerID); err != nil {
+		logrus.Infof("failed to delete peerID(%s) from peerProgress: %v", peerID, err)
+		return false
+	}
+	return true
+}
+
+func (pm *Manager) deletePeerIDFromBlackInfo(ctx context.Context, peerID string) bool {
+	result := true
+	// delete the black info which use peerID as the key
+	pm.clientBlackInfo.Delete(peerID)
+
+	// TODO: is it necessary?
+	keys := pm.clientBlackInfo.ListKeyAsStringSlice()
+	for _, key := range keys {
+		mmap, err := pm.clientBlackInfo.GetAsMap(key)
+		if err != nil {
+			result = false
+			logrus.Infof("failed to delete peerID(%s) from peer(%s) client black info: %v", peerID, key, err)
```

Comment:
We hardly use Info log level for an error message. Maybe Warn level is a little bit better.